templates/profile.php

```diff
@@ -204,10 +206,15 @@ function pmpromd_profile_shortcode($atts, $content=null, $code="")
 	if(empty($user_id) && !empty($_REQUEST['pu']))
 	{
 		//Get the profile user
-		if(is_numeric($_REQUEST['pu']))
-			$pu = get_user_by('id', $_REQUEST['pu']);
-		else
-			$pu = get_user_by('slug', $_REQUEST['pu']);
+		$user_identifier = apply_filters( 'pmpromd_user_identifier', 'user_nicename' );
+
+		if( !empty( $_REQUEST['pu'] ) ) {
+			$pu = get_user_by( $user_identifier, $_REQUEST['pu'] );
+		} else if( !empty($current_user->ID ) ) {
+			$pu = $current_user;
+		} else {
+			$pu = false;
+		}
 
 		$user_id = $pu->ID;

```

Comment:
We need to check if `$pu` was found (it could be `false`), otherwise this line would cause PHP errors.